chain/jsonrpc/src/lib.rs

```diff
@@ -261,25 +265,47 @@ impl JsonRpcHandler {
                     .await;
                 match tx_status_result {
                     Ok(Ok(Some(outcome))) => break Ok(outcome),
-                    Ok(Ok(None)) => {}
+                    Ok(Ok(None)) | Ok(Err(TxStatusError::MissingTransaction(_))) => {
+                        if let TransactionInfo::Transaction(tx) = &tx_info {
+                            if let Ok(NetworkClientResponses::InvalidTx(e)) =
+                                self.send_tx(tx.clone(), true).await
+                            {
+                                if let InvalidTxError::InvalidNonce { .. } = e {
+                                    if nonce_error.is_none() {
+                                        nonce_error = Some(e);
+                                    }
+                                } else {
+                                    break Err(TxStatusError::InvalidTx(e));
+                                }
+                            }
+                        }
+                        if let Ok(Err(e)) = tx_status_result {
+                            break Err(e);
+                        }
+                    }
                     Ok(Err(err)) => break Err(err),
                     Err(_) => break Err(TxStatusError::InternalError),
                 }
                 let _ = delay_for(self.polling_config.polling_interval).await;
             }
         })
         .await
-        .map_err(|_| TxStatusError::TimeoutError)?
+        .map_err(|_| {
+            if let Some(e) = nonce_error {
```

Comment:
@ilblackdragon I am not sure what's the best way to handle this. We cannot return it inside the loop since we poll quote frequently and the state will have been updated before the transaction is fully complete.